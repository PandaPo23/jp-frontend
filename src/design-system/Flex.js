import system from '../utils/System';
import Box from './Box';

const Flex = system(
  'Flex',
  {
    extend: Box,
    display: 'flex',
  },
  ({ centered }) => ({
    justifyContent: centered && 'center',
    alignItems: centered && 'center',
  })
);

export default Flex;
