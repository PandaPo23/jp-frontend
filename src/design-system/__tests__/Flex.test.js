import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Flex from '../Flex';

describe('Flex component', () => {
  it('should render as Flex', () => {
    const wrapper = mount(<Flex />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render centered Flex', () => {
    const localProps = {
      centered: true,
    };
    const wrapper = mount(<Flex {...localProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(
      wrapper
        .find(Flex)
        .first()
        .prop('centered')
    ).toEqual(true);
  });
});
