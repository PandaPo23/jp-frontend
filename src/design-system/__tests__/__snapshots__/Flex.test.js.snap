// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex component should render as Flex 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<Flex
  display="flex"
>
  <div
    className="emotion-0"
  />
</Flex>
`;

exports[`Flex component should render centered Flex 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<Flex
  centered={true}
  display="flex"
>
  <div
    className="emotion-0"
  />
</Flex>
`;
